### PR TITLE
feat: Display view switcher for recent files

### PR DESCRIPTION
**Pull request by doubleface**
And also allow to hide sorter when canSort is not true.

image

## Summary by CodeRabbit

* **Bug Fixes**
  * Mobile file list header layout corrected to preserve right-side alignment while properly showing sort controls.

* **New Features**
  * Centralized file-sorting applied across Folder and Recent views for consistent, stable sorting and selection behavior.

* **Tests**
  * Added comprehensive tests covering sorting logic, default and external sort orders, secondary sorting, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/modules/filelist/FileListHeaderMobile.jsx [hunk 1]
```diff
@@ -36,13 +37,14 @@ const FileListHeaderMobile = ({
     [setIsShowingSortMenu]
   )
 
-  if (!canSort) return null
+  if (!canSort && !canSwitchViewType) return null
```
Comment (rezk2ll) [inline src/modules/filelist/FileListHeaderMobile.jsx]: i don't think we need to have this `canSwitchViewType` here, no matter what mobile or desktop we should be able to switch the view

File: src/modules/filelist/FileListHeaderMobile.jsx [hunk 2]
```diff
@@ -63,6 +65,7 @@ const FileListHeaderMobile = ({
           />
         )}
         <TableHeader
+          style={{ visibility: canSwitchViewType ? 'visible' : 'hidden' }}
```
Comment (rezk2ll) [inline src/modules/filelist/FileListHeaderMobile.jsx]: it should be always displayed

**Comment by rezk2ll** [comment]
i don't understand why this PR is needed

File: src/modules/filelist/FileListHeaderMobile.jsx (hunk 2 above)
Comment (doubleface) [reply src/modules/filelist/FileListHeaderMobile.jsx]: Yes, but at first the only criteria to display the header was if canSort is thruthy. Here, for the recents view, I want to display the header but only with the "change view type" button. 
Yes, at the moment, "change view type" button will always be display, but canSwitchViewType gives me a new criteria to display the mobile header or not and allowing to hide it make the api more coherent I think.

Comment (doubleface) [reply src/modules/filelist/FileListHeaderMobile.jsx]: Or should we change the logic and add a "hideHeader" prop to FolderViewBody ?

**Comment by doubleface** [comment]
> i don't understand why this PR is needed

This PR is needed to display the view switcher on the recents file view.

I am sorry because at first I also added the sort feature because it was the only way to display the toolbar.
After some thinking, I prefered to make it possible to display only the view switcher in the toolbar, without the sorting.

File: src/modules/views/Folder/hooks/useFileSorting.spec.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  })
+
+  it('should apply both primary and secondary sort', () => {
+    const { result } = renderHook(() =>
+      useFileSorting('folder-1', mockQueryResults, mockOrderProps)
+    )
+
+    // Should be sorted by name in descending order, then by type
+    expect(result.current.sortedFiles).toHaveLength(3)
+  })
```
Comment (doubleface) [inline src/modules/views/Folder/hooks/useFileSorting.spec.js:144]: Added some tests on sorting

File: src/modules/filelist/FileListHeaderMobile.jsx (hunk 2 above)
Comment (doubleface) [reply src/modules/filelist/FileListHeaderMobile.jsx]: Let's always show the header with the view switch and hide the sorter when canSort is not true

File: src/modules/views/Folder/hooks/useFileSorting.spec.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    const fileInfo = sortedFiles.map(file => ({
+      name: file.name,
+      type: file.type
+    }))
+    expect(fileInfo).toEqual([
+      { name: 'folder-c', type: 'directory' },
+      { name: 'file-b.txt', type: 'file' },
+      { name: 'file-a.txt', type: 'file' }
+    ])
+  })
```
[the remark this answers is not in the document]
Comment (doubleface) [reply src/modules/views/Folder/hooks/useFileSorting.spec.js:144]: done

File: src/modules/views/Folder/hooks/useFileSorting.spec.js [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+          }
+        ],
+        {}
+      )
+    )
+
+    // Secondary sort should put directories before files
+    expect(result.current.sortedFiles[0].type).toBe('directory')
+    expect(result.current.sortedFiles[1].type).toBe('file')
+  })
```
[the remark this answers is not in the document]
Comment (doubleface) [reply src/modules/views/Folder/hooks/useFileSorting.spec.js]: done

**Review by rezk2ll — APPROVED** [review]
